torchinductor/lowering.py

```diff
@@ -693,6 +721,96 @@ def inner_fn(index):
     )
 
 
+@register_lowering(aten.select_scatter)
+def select_scatter(x, src, dim: int, index: int):
+    x_loader = x.make_loader()
+    dim = _validate_dim(x, dim, 0)
+    src = expand(unsqueeze(src, dim), x.get_size())
+    src_loader = src.make_loader()
+
+    def inner_fn(idx):
+        return ops.where(
+            ops.eq(
+                ops.index_expr(idx[dim], torch.int32),
+                ops.constant(index, torch.int32),
+            ),
+            src_loader(idx),
+            x_loader(idx),
+        )
+
+    return Pointwise.create(
+        device=x.get_device(),
+        dtype=x.get_dtype(),
+        inner_fn=inner_fn,
+        ranges=list(x.get_size()),
+    )
+
+
+@register_lowering(aten.slice_scatter)
+def slice_scatter(x, src, dim=0, start=None, end=None, step=1):
```

Comment:
THat's probably not the most efficient decomposition (an efficient one would take a slice of the tensor with the new strides and sizes, and copy the `src` tensor to it, similar to aten implementation https://fburl.com/code/2ne9ady3. Since you already have a slice_ lowering, that should be something like clone + slice_ + copy?

Reply:
Ah interesting.  You think two kernels would be better.  I had assumed it would be memory bound, so this would be faster.

Reply:
I don't know for sure, your way indeed might turn out faster. 
Where is slice_scatter coming from? Are you running functionalization pass? (Because I thought it's used in functionalization only)

Reply:
This is coming from AOT Autograd, which I believe does functionalization.  We may want to turn this off, as I worry this will de-optimize things.

Reply:
I looked a bit more into this.  In this case `x` is an unrealized1 zero tensor, so I think this decomp might be fine since `x_loader(...)` just gets mapped to `0`.

I think we might need to take a pass over the decomps with performance in mind though.  The decomps are de-inplacing things, then in codegen we are re-inplacing things -- sometimes creating extra copies in output code.